File: domain/src/main/java/mixter/domain/identity/handlers/RegisterSession.java
package mixter.domain.identity.handlers;

import mixter.domain.identity.SessionProjection;
import mixter.domain.identity.SessionStatus;
import mixter.domain.identity.events.UserConnected;

public class RegisterSession {
    private SessionProjectionRepository sessionProjectionRepository;

    public RegisterSession(SessionProjectionRepository sessionProjectionRepository) {
        this.sessionProjectionRepository = sessionProjectionRepository;
    }

    public void apply(UserConnected event) {
        sessionProjectionRepository.save(new SessionProjection(event.getSessionId(), event.getUserId(), SessionStatus.CONNECTED));
    }
}

File: domain/src/main/scala/mixter/domain/identity/handlers/RegisterSession.scala
package mixter.domain.identity.handlers

import mixter.domain.identity.event.{UserConnected, UserDisconnected}
import mixter.domain.identity.{SessionProjection, SessionProjectionRepository, SessionStatus}

class RegisterSession(repository:SessionProjectionRepository){
  def apply(event: UserConnected):Unit = {
    val projection = SessionProjection(event.sessionId,  event.id, SessionStatus.CONNECTED)
    repository.save(projection)
  }
  def apply(event: UserDisconnected):Unit = {
    val projection = SessionProjection(event.sessionId,  event.id, SessionStatus.DISCONNECTED)
    repository.replaceBy(projection)
  }
}
